api/README.md

```diff
@@ -0,0 +1,218 @@
+approach. Note that it makes the assumption that we're building for the Kia
+Soul Petrol with the `"-DKIA_SOUL=ON"`.
+
+```
+cmake_minimum_required(VERSION 2.8)
+project(an_example)
+set(OSCC_API_INSTALL /path_to/oscc_directory/api)
+set(OSCC_INCLUDES ${OSCC_API_INSTALL}/include ${OSCC_API_INSTALL}/src)
+set(SOURCES ${CMAKE_SOURCE_DIR}/src/main.c ${OSCC_API_INSTALL}/src/oscc.c)
+add_executable(${PROJECT_NAME} ${SOURCES})
+target_compile_definitions(${PROJECT_NAME} PUBLIC "-DKIA_SOUL=ON")
+target_include_directories(${PROJECT_NAME} PUBLIC ${OSCC_INCLUDES})
+```
+
+## Using the API
+
+Let's walk through actually writing some code that uses the OSCC API, an example
+program the listens to all the information that OSCC reports. Feel free to
+experiment with this code, if it's all put into a `main.c` you can use
+any of the CMakeLists.txt options above to build it.
+
+First, we'll do some generic setup. This isn't a requriement of the OSCC API
+but might make our example a little more useful.
+
+```
+#include <stdio.h> // printf
+#include <signal.h> // SIGINT
+#include <unistd.h> // usleep
+```
+Okay, now for the actual OSCC API include that allows us to initiate,
+listen to each message type and then properly release OSCC. `oscc.h` is
+all you need!
+
+```
+#include "oscc.h" // oscc_open, oscc_disable, oscc_close and each subscribe.
+```
+In order for our example to trigger the proper cleanup when we press `Ctrl+C`
+to quit the program, we'll setup a signal handler to let us know.
+
+```
+// Global indication of Ctrl+C.
+static int EXIT_SIGNALED = OSCC_OK;
+
+// Help exiting gracefully.
+void signal_handler(int signal_number)
+{
+    if (signal_number == SIGINT)
+    {
+        EXIT_SIGNALED = OSCC_ERROR;
+    }
+}
+```
+
+Now lets walk through our OSCC message subscribers. When we get a report from
+something OSCC provides, let's print a message reporting what we see.
+
+__Throttle__
+```
+void throttle_callback(oscc_throttle_report_s * report)
+{
+    printf("Hello from throttle_callback!\n");
+}
+```
+__Steering__
+```
+void steering_callback(oscc_steering_report_s * report)
+{
+    printf("steering_callback reporting!\n");
+}
+```
+__Brake__
+```
+void brake_callback(oscc_brake_report_s * report)
+{
+    printf("'Brake-ing' news from the brake_callback!\n");
+}
+```
+__Fault__
+```
+void fault_callback(oscc_fault_report_s * report)
+{
+    printf("Oh! Something caused a fault and we made it to the fault_callback. We're seeing a ");
+
+    if (report->fault_origin_id == FAULT_ORIGIN_BRAKE)
+    {
+        printf("brake fault!\n");
+    }
+    else if (report->fault_origin_id == FAULT_ORIGIN_STEERING)
+    {
+        printf("steering fault!\n");
+    }
+    else if (report->fault_origin_id == FAULT_ORIGIN_THROTTLE)
+    {
+        printf("throttle fault!\n");
+    }
+}
+```
+__OBD__
+```
+static void obd_callback(struct can_frame * frame)
+{
+    printf("From the obd_callback, CAN ID: %u\n", frame->can_id);
```

Comment:
Non-blocker nit: `%x` might be nice here since CAN IDs are almost always specified in hex.